tests/functional/test_install.py

```diff
@@ -1267,9 +1267,17 @@ def test_install_pep508_with_url_in_install_requires(script):
             'ce1a869fe039fbf7e217df36c4653d1dbe657778b2d41709593a0003584405f4'
         ],
     )
-    res = script.pip('install', pkga_path, expect_error=True)
-    assert "Direct url requirement " in res.stderr, str(res)
-    assert "are not allowed for dependencies" in res.stderr, str(res)
+    res = script.pip('install', pkga_path)
+    assert "Successfully installed packaging-15.3" in str(res), str(res)
+
+
+@pytest.mark.network
+def test_install_from_pypi_with_ext_url_in_install_requires_is_blocked(script):
+    res = script.pip('install', '-vvv', 'pep-508-url-deps', expect_error=True)
+    assert "Packages installed from PyPI cannot " in res.stderr, str(res)
```

Comment:
nit: Instead of checking the entire message over multiple statements, I'd rather that you compose an `expected` string and then just do `assert expected in result.stderr`.

We should also check the exit code here.